### PR TITLE
BUG: isin fails to correctly compare None values.

### LINKED ISSUE #35565: different behaviour of df.isin() in 1.0.5/1.1.0, when df contains None
```
>>> import pandas as pd
>>> x = pd.DataFrame([['foo', 'bar'], [1,None]])
>>> y = x[1].copy()

# PANDAS 1.0.5
>>> x.isin(y)
       0     1
0  False  True
1  False  True

# PANDAS 1.1.0
>>> x.isin(y)
       0      1
0  False   True
1  False  False

```

Is this change intended?

### PR DESCRIPTION
**Pull request by DriesSchaumont**
- [x] closes #35565 
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/common.py [hunk 1]
```diff
@@ -478,14 +478,15 @@ def get_rename_function(mapper):
     if isinstance(mapper, (abc.Mapping, ABCSeries)):
 
         def f(x):
-            if x in mapper:
-                return mapper[x]
-            else:
+            if isinstance(x, bool):
```
Comment (jreback) [inline pandas/core/common.py:481]: why are you changing this?

File: pandas/_libs/ops.pyx [hunk 2]
```diff
@@ -160,17 +160,19 @@ def vec_compare(ndarray[object] left, ndarray[object] right, object op) -> ndarr
         for i in range(n):
             x = left[i]
             y = right[i]
-
-            if checknull(x) or checknull(y):
+            if x is None and y is None:
```
Comment (jreback) [inline pandas/_libs/ops.pyx:163]: this is superfluous as checknull already checks None

File: pandas/_libs/ops.pyx [hunk 3]
```diff
@@ -160,17 +160,19 @@ def vec_compare(ndarray[object] left, ndarray[object] right, object op) -> ndarr
...
                 result[i] = True
             else:
                 result[i] = PyObject_RichCompareBool(x, y, flag)
     else:
         for i in range(n):
             x = left[i]
             y = right[i]
-
-            if checknull(x) or checknull(y):
+            if x is None and y is None:
```
Comment (jreback) [inline pandas/_libs/ops.pyx:173]: this is contrary to what we do with nulls generally, does anything break when you make this change (only)?

File: pandas/tests/frame/methods/test_rename.py [hunk 4]
```diff
@@ -403,3 +403,13 @@ def test_rename_with_duplicate_columns(self):
             ],
         ).set_index(["STK_ID", "RPT_Date"], drop=False)
         tm.assert_frame_equal(result, expected)
+
+
+def test_rename_dict_wrong_matches():
```
Comment (jreback) [inline pandas/tests/frame/methods/test_rename.py:408]: this is orthgonal?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/test_algos.py [hunk 5]
```diff
@@ -1070,6 +1070,15 @@ def test_isin_float_df_string_search(self):
         expected_false = DataFrame({"values": [False, False]})
         tm.assert_frame_equal(result, expected_false)
 
+    def test_isin_none(self):
+        # GH35565
+        x = DataFrame([["foo", "bar"], [1, None]])
+        y = x[1].copy()
+        res = x.isin(y)
+        expected = DataFrame([[False, True], [False, True]])
+        print(expected)
```
Comment (jbrockmendel) [inline pandas/tests/test_algos.py:1079]: extraneous print

**Comment by jbrockmendel** [comment]
@DriesSchaumont can you respond to jreback comments

**Comment by DriesSchaumont** [comment]
Hi @jbrockmendel, I am in the middle of moving to a new place and I currently have limited internet access. I will try to come back to this PR as soon as possible. I hope this is not blocking anything?

**Comment by jbrockmendel** [comment]
> I hope this is not blocking anything?

no, not to worry

**Comment by DriesSchaumont** [comment]
Seems I accidentally pushed a commit that was meant to be in another branch. Updating this PR now.

File: pandas/_libs/ops.pyx (hunk 2 above)
Comment (DriesSchaumont) [reply pandas/_libs/ops.pyx:160]: Yep, this was incorrect. Changed.

File: pandas/_libs/ops.pyx (hunk 3 above)
Comment (DriesSchaumont) [reply pandas/_libs/ops.pyx:170]: AFAIK, this does not break any tests. However, I am starting to think that #35565 is not actually a bug, but intended behavior. From the [docs](https://pandas.pydata.org/pandas-docs/stable/user_guide/missing_data.html#values-considered-missing): "One has to be mindful that in Python (and NumPy), the nan's don’t compare equal, but None's do. Note that pandas/NumPy uses the fact that np.nan != np.nan, and treats None like np.nan."

I see the same behavior in `scalar_compare`:
https://github.com/pandas-dev/pandas/blob/1269bc6b7f3f3f4fc253cfa11f795d6bccda55a0/pandas/_libs/ops.pyx#L74-L86

I found this related issue: https://github.com/pandas-dev/pandas/issues/34975

Should we close this PR and the issue, then?

File: pandas/_libs/ops.pyx (hunk 2 above)
Comment (jbrockmendel) [reply pandas/_libs/ops.pyx:160]: what was incorrect?

rule of thumb: let the question-asker mark the conversation as resolved

**Comment by jbrockmendel** [comment]
if possible we should reuse libmissing.is_matching_na

**Comment by jreback** [comment]
closing as stale, if you want to continue working, please ping.

**Pull request closed without merging**